Makefile

```diff
@@ -162,13 +162,45 @@ quiet = $(if $(V),$($(1)),$(call output,$1,$@);$($(1)))
 this: $(libceed) $(ceed.pc)
 # run 'this' target in parallel
 all:;@$(MAKE) $(MFLAGS) V=$(V) this
+info:
+	$(info ------------------------------------)
+	$(info CC        = $(CC))
+	$(info FC        = $(FC))
+	$(info CPPFLAGS  = $(CPPFLAGS))
+	$(info CFLAGS    = $(value CFLAGS))
+	$(info FFLAGS    = $(value FFLAGS))
+	$(info NVCCFLAGS = $(value NVCCFLAGS))
+	$(info LDFLAGS   = $(value LDFLAGS))
+	$(info LDLIBS    = $(LDLIBS))
+	$(info OPT       = $(OPT))
+	$(info AFLAGS    = $(AFLAGS))
+	$(info ASAN      = $(or $(ASAN),(empty)))
+	$(info V         = $(or $(V),(empty)) [verbose=$(if $(V),on,off)])
+	$(info ------------------------------------)
+	$(info CUDA_DIR  = $(CUDA_DIR))
+	$(info OCCA_DIR  = $(OCCA_DIR))
+	$(info MAGMA_DIR = $(MAGMA_DIR))
```

Comment:
Is it intentional that these currently print even if the directory doesn't exist and was only defined by `?=` in the makefile?

Reply:
+1 to this. I think it's best to print these directories only when they are used by libCEED.

Reply:
Yes, printing these `*_DIR` variables is absolutely intentional - they are used to control the build by specifying where to search for external dependencies.

Reply:
Ah, I had thought that the purpose of the printout was to give the location of the dependencies of the currently built version of libCEED.

Reply:
It is just confusing to print `MAGMA_DIR = ../magma` when that directory does not exist and there is no attempt to use it.

Reply:
Actually, the contents of the variable `MAGMA_DIR` is always used to check for MAGMA, here:
https://github.com/CEED/libCEED/blob/0948660574688e7f772302e3ec183ecf15d81fe4/Makefile#L178
The idea for `make info` is to show most of the variables that control the build - that way a developer, or a user, can quickly get a summary of what can be adjusted on the command line.